src/generate.py

```diff
@@ -751,14 +754,14 @@ def generate_C_epilogue(c_file, prefix):
     char errbuf[1024];
     if (content == NULL) {
         if (asprintf (err, "cannot read the file: %%s", filename) < 0)
-            *err = "error allocating memory";
+            *err = strdup("error allocating memory");
```

Comment:
I think on OOM we should just print an error and exit, what do you think?  If we try to use strdup after an OOM, it is more probably we hit it again, can be:
```
fprintf (stderr, "OOM\n");
exit (EXIT_FAILURE);
```

We can slowly adapt other calls, or even better we could have wrappers like in:

https://github.com/giuseppe/crun/blob/master/src/libcrun/utils.c#L108-L136

where OOM is:

https://github.com/giuseppe/crun/blob/master/src/libcrun/error.c#L61-L65

Reply:
I don't know the implement of `asprintf`, this is the man page of `asprintf`:
```
When successful, these functions return the number of bytes printed, just like sprintf(3).  If memory allocation wasn't possible, or some other error occurs, these functions will return -1, and the contents of strp are undefined.
```
It can be other error occurs, and if `strdup` is failed, the `err` still be `NULL`, we will not call `free`.

Besides, If we run this function in daemon process, we'd better call `exit` by its caller, maybe it have some exit handler, like what you do in `libcrun`.

Reply:
that is fine, the problem is that we are trying again to use `strdup` in a situation when there is no memory left to allocate.  It is probably fine to return NULL in this case though.